satpy/readers/__init__.py

```diff
@@ -475,14 +477,29 @@ def load_readers(filenames=None, reader=None, reader_kwargs=None,
     else:
         remaining_filenames = set(filenames or [])
 
+    reader_kwargs = reader_kwargs or {}
+
+    # ensure one reader_kwargs per reader, None if not provided
+    if reader is None:
+        reader_kwargs = {None: reader_kwargs}
+    elif reader_kwargs.keys() != set(reader):
+        reader_kwargs = dict.fromkeys(reader, reader_kwargs)
+
+    reader_kwargs_without_filter = {}
+    for (k, v) in reader_kwargs.items():
+        reader_kwargs_without_filter[k] = v.copy()
+        reader_kwargs_without_filter[k].pop('filter_parameters', None)
+
```

Comment:
Would it be possible to move this whole `reader_kwargs` thing into a separate method just for cleanliness?

Also, if `reader` was changed to a list could the below changes be made to use `[reader[idx]]` instead of the ternary operation currently in there?

Reply:
Move, yes, will do.

Avoiding ternary, difficult.  If `reader is None`, `configs_for_reader` loops through all possible readers.  For `reader[idx]` to work in this case it should either be a sequence with all readers or some pseudo-sequence object doing magic, and `reader_kwargs_without_filter` would need to understand all readers.  It seems unlikely that a user would not pass any readers but still pass readers as keys to `reader_kwargs`.

Reply:
Ok on the ternary. :+1: